addons/account/models/account_payment.py

```diff
@@ -394,16 +394,18 @@ def _compute_company_id(self):
             if payment.journal_id.company_id not in payment.company_id.parent_ids:
                 payment.company_id = (payment.journal_id.company_id or self.env.company)._accessible_branches()[:1]
 
-    @api.depends('invoice_ids.payment_state')
+    @api.depends('invoice_ids.payment_state', 'move_id.line_ids.amount_residual')
     def _compute_state(self):
         for payment in self:
             if not payment.state:
                 payment.state = 'draft'
-            if (
-                not payment.outstanding_account_id
-                and payment.invoice_ids
-                and all(invoice.payment_state == 'paid' for invoice in payment.invoice_ids)
-            ):
+            if payment.outstanding_account_id:
+                move = payment.move_id
+                lines = move.line_ids.filtered(lambda l: l.account_id.account_type in ('asset_receivable', 'liability_payable'))
+                amount_residual = sum(lines.mapped('amount_residual'))
+                if move and move.currency_id.is_zero(amount_residual):
+                    payment.state = 'paid'
+            elif payment.invoice_ids and all(invoice.payment_state == 'paid' for invoice in payment.invoice_ids):
```

Comment:
@william-andre So, after the change applied in this commit, the following problems show up (and I am not sure what to do because I don't fully understand the functional requirements):
1. In the UI, after creating a partial payment on an invoice with an outstanding account id (its payment method has account id) the payment get automatically paid (with no reconciliation done at all).
2. The unit test fails when the `accountant` module is not installed in the DB, should I move the unit tests to the `accountant` module? This probably happens because of the different definitions of `_get_invoice_in_payment_state` in `account` and `accountant` but I am not quite following why it affects the state of the payment.

Reply:
1. So it means the amount residual is 0? That's wrong, as you probably guessed. Is it only in the UI?
2. You can use a mock to control `_get_invoice_in_payment_state`
You should probably be aware of this incoming change as well https://github.com/odoo/odoo/pull/182390 where something similar/related is done

Reply:
@william-andre
1. Yeah, I traced the code and the `amount_residual` = 0. Maybe the lines here 
```python
lines = move.line_ids.filtered(lambda l: l.account_id.account_type in ('asset_receivable', 'liability_payable'))
```
should have the outstanding account type as well in the filter?
2. ok I will test that :)